staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -89,9 +91,13 @@ type ServerRunOptions struct {
 	// This grace period is orthogonal to other grace periods, and
 	// it is not overridden by any other grace period.
 	ShutdownWatchTerminationGracePeriod time.Duration
+
+	// FeatureGate are the featuregate to install on the CLI
+	FeatureGate      featuregate.MutableVersionedFeatureGate
```

Comment:
recording the feature gates to use is nice, can we give this the non-mutable interface to read from, and put the responsibility for binding flags / calling SetEmulationVersion on the NewServerRunOptions caller?